### PR TITLE
New command: unpublish posts

### PR DESCRIPTION
**Pull request by imathis**
This will make it possible to convert posts back to drafts.

It makes sense for people to add the `published: false` method to posts to prevent them from being published, but in a world where Jekyll has some nifty draft features, it would be nicer to just convert them to a draft. This command makes it easy to do that. There are two ways to use the `unpublish` command:

```sh
$ octopress unpublish "some post"
$ octopress unpublish --path _posts/2015-01-10-some-post.md
```

Here are the options:

| Option             | Description                               |
|:-------------------|:------------------------------------------|
| `--path`           | Specify a path instead of using search    |
| `--force`          | Overwrite existing file.                  |

If you have multiple posts that match the search string, you will be prompted to chose a post. That will look like this:

```
$ octopress unpublish random
Found 2 posts matching 'random'
  1) 2011-11-11-a-random-post.markdown
  2) 2012-11-11-another-random-post.markdown
Which do you want to unpublish? (enter a number): 
```

And that's pretty much it.

Hey @parkr, I'd love to have your eyes on this one.

### CODE REVIEW

**Comment by parkr** [comment]
I think a path might be a better default, using `--title "My post"` as a backup?

**Comment by imathis** [comment]
I had it that way at first, but typing a path to a post is really tedious. `_p[tab]2015-[tab]-so[tab]` especially if you're a prolific writer. But yeah I like how it's so exact to uses path.

The other option is that I could do both. The search will work with the path just as well as a search string

**Comment by parkr** [comment]
Like `File.exist?(arg) ? unpublish(arg) : unpublish(post_from_title(arg))`?

**Comment by imathis** [comment]
Yep. That's right @parkr. That's basically what I've done now. And to be consistent, I've added this feature to `publish` and `isolate` as well. Now you can very easily convert to and from draft state and isolate posts to speed up generation. It feels pretty nice to be able to use these commands this way.

File: assets/docs/index.md [hunk 1]
```diff
@@ -163,19 +182,19 @@ File name extensions are unnecessary since they're just plain text anyway.
...
-
-This will copy all other posts into `_posts/_exile` where they will be ignored by Jekyll during the build process. Here are some examples:
-
-- `octopress isolate` isolates the most recently dated post.
-- `octopress isolate cats` isolates all posts with the word 'cats' in the filename.
-- `octopress isolate --path _posts/2014-10-11-kittens.md` isolates the post at the given path.
 
-To reintegrate all exiled posts, run `octopress integrate` which will restore all posts from `_posts/_exile` to `_posts`.
+In the first example, `octopress isolate` without any arguments will send all posts to `_posts/_exile`. The second example demonstrates
+how to isoalte a post by passing a path. The third example will isolate a post which matches the search string. If multiple posts match,
```
Comment (parkr) [inline assets/docs/index.md]: `how to isolate`

File: assets/docs/index.md [hunk 2]
```diff
@@ -163,19 +182,19 @@ File name extensions are unnecessary since they're just plain text anyway.
...
+you will be prompted to chose from a menue which post to isolate.
```
Comment (parkr) [inline assets/docs/index.md]: `menu`

**Comment by parkr** [comment]
Niiiice!!

**Pull request merged**